core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -103,4 +104,16 @@ package object config {
     .stringConf
     .checkValues(Set("hive", "in-memory"))
     .createWithDefault("in-memory")
+
+  private[spark] val LISTENER_BUS_EVENT_QUEUE_SIZE =
+    ConfigBuilder("spark.scheduler.listenerbus.eventqueue.size")
+      .intConf
+      .transform((x: Int) => {
```

Comment:
Agreed. The method is intended for something else. Having the validator functionality here would be good to do so. I will make the changes and fix the tests.